app/src/main/java/com/example/schedule/HtmlParser.kt

```diff
@@ -7,38 +7,75 @@ import org.jsoup.nodes.Document
 import org.jsoup.select.Elements
 import java.util.*
 import com.example.schedule.ScheduleDataBase.*
+import org.jsoup.nodes.Element
+import kotlin.collections.ArrayList
 
 class HtmlParser{
 
     fun getHtmlFromWeb(db : ScheduleDataBase) {
-        val groups: ArrayList<String> = arrayListOf()
-        val directions: ArrayList<String> = arrayListOf()
         Thread{
+            db.clear_directions_table()
+            db.clear_lessons_table()
             try {
                 val doc: Document = Jsoup.connect("https://jjpega.pythonanywhere.com/get").get()
-                val group: Elements = doc.select("p[id=\"group\"]")
-                for (index in group.indices) {
-                    groups.add(group[index].text())
-                }
-                groups.distinct()
 
-                val direction: Elements = doc.select("td[id=\"direction\"]")
-                for (index in direction.indices) {
-                    directions.add(direction[index].text())
+                var directions: ArrayList<String> = arrayListOf()
+                var i = 0
+                while (true){
+                    try {
+                        val direction: String = doc.select("td[id=\"direction\"]")[i].text()
+                        directions.add(direction)
+                        i += 1
+                    }
+                    catch (e: Exception) {
+                        break
+                    }
                 }
-                directions.distinct()
 
-                for (index in groups.indices) {
-                    db.group_add(groups[index], directions[index])
+                var num_directions: Int = directions.size
+                i = 0
+                while (i <= num_directions){
```

Comment:
```suggestion
                while (i < num_directions){
```
Если в массиве 10 элементов, и ты считаешь их от нуля, то это 0 1 2 3 4 5 6 7 8 9 (здесь 10 чисел, число "10" не нужно).